packages/plugin-types/API.md

```diff
@@ -142,6 +142,9 @@ export type ConstructFactoryGetInstanceProps = {
     resourceNameValidator?: ResourceNameValidator;
 };
 
+// @public (undocumented)
+export type DataLogLevel = 'none' | 'error' | 'info' | 'debug' | 'all';

```

Comment:
This type should be moved to `backend-data` .
Similarly to https://github.com/aws-amplify/amplify-backend/blob/6dd93d17fd04df3bcc2984682a2aa6b383c9a445/packages/backend-function/src/factory.ts#L70 . The function just aliases it because it uses all values.

The `LogLevel` defined in platform types is superset of known log levels.

We can then define `DataLogLevel` either like you have it here. 
OR we can do something like this `export type DataLogLevel = Extract<LogLevel, 'info' | 'debug'>;` to assure that `export type DataLogLevel = Extract<LogLevel, 'info' | 'debug'>;` is indeed subset of `LogLevel` .